core/crypto/crypto_store/key_type.hpp

```diff
@@ -17,52 +18,158 @@ namespace kagome::crypto {
+
+    inline constexpr auto operator<=>(const KeyType &other) const {
+      return id_ <=> other.id_;
+    }
+
+    friend inline ::scale::ScaleEncoderStream &operator<<(
+        ::scale::ScaleEncoderStream &s, const KeyType &v) {
+      return s << v.id_;
+    }
+
+    friend inline ::scale::ScaleDecoderStream &operator>>(
+        ::scale::ScaleDecoderStream &s, KeyType &v) {
+      return s >> v.id_;
+    }
+
+   private:
+    uint32_t id_{0};
   };
 
-  /**
-   * @brief makes string representation of KeyTypeId
-   * @param key_type_id KeyTypeId
-   * @return string representation of KeyTypeId
-   */
-  std::string encodeKeyTypeIdToStr(KeyTypeId key_type_id);
+  struct KeyTypes {
+    /// Key type for Babe module, built-in.
+    static constexpr KeyType BABE = "babe"_key;
+    /// Key type for Sassafras module, built-in.
+    static constexpr KeyType SASSAFRAS = "sass"_key;
+    /// Key type for Grandpa module, built-in.
+    static constexpr KeyType GRANDPA = "gran"_key;
+    /// Key type for controlling an account in a Substrate runtime, built-in.
+    static constexpr KeyType ACCOUNT = "acco"_key;
+    /// Key type for Aura module, built-in.
+    static constexpr KeyType AURA = "aura"_key;
+    /// Key type for BEEFY module.
+    static constexpr KeyType BEEFY = "beef"_key;
+    /// Key type for ImOnline module, built-in.
+    static constexpr KeyType IM_ONLINE = "imon"_key;
+    /// Key type for AuthorityDiscovery module, built-in.
+    static constexpr KeyType AUTHORITY_DISCOVERY = "audi"_key;
+    /// Key type for staking, built-in.
+    static constexpr KeyType STAKING = "stak"_key;
+    /// A key type for signing statements
+    static constexpr KeyType STATEMENT = "stmt"_key;
+    /// A key type ID useful for tests.
+    static constexpr KeyType DUMMY = "dumy"_key;
+
+    static constexpr KeyType KEY_TYPE_ASGN = "asgn"_key;
+    static constexpr KeyType KEY_TYPE_PARA = "para"_key;
+
+    KeyTypes() = delete;
+
+    static constexpr bool is_supported(const KeyType &key_type) {
+      switch (key_type) {
+        case KeyTypes::BABE:
+        case KeyTypes::SASSAFRAS:
+        case KeyTypes::GRANDPA:
+        case KeyTypes::ACCOUNT:
+        case KeyTypes::AURA:
+        case KeyTypes::BEEFY:
+        case KeyTypes::IM_ONLINE:
+        case KeyTypes::AUTHORITY_DISCOVERY:
+        case KeyTypes::STAKING:
+        case KeyTypes::STATEMENT:
+        case KeyTypes::DUMMY:
+        case KeyTypes::KEY_TYPE_ASGN:
+        case KeyTypes::KEY_TYPE_PARA:
+          return true;
+      }
+      return false;
+    }
+  };
 
   /**
-   * @brief restores KeyTypeId from its string representation
-   * @param param string representation of key type
-   * @return KeyTypeId
+   * @brief makes string representation of KeyType
+   * @param key_type KeyType
+   * @return string representation of KeyType
    */
-  KeyTypeId decodeKeyTypeIdFromStr(std::string_view str);
+  std::string encodeKeyTypeToStr(const KeyType &key_type);
```

Comment:
I think they're all string literals, why not return string_view?

Reply:
Returning reference is unsafe (may return `std::array<char, 4>`)

Reply:
string_view should have longer lifetime. For unknown type, i.e. based on received data (temporary).

Reply:
Ah, I thought it transformed the contexpr KeyTypes.